pyproject.toml

```diff
@@ -0,0 +1,547 @@
+[tool.black]
```

Comment:
@iarspider , this has nothing to do with the bot changes ..... right? I would suggest to open a separate PR for this. By the way what is the purpose of this file?

Reply:
Sure, I will split it out. In general, this file describes how a Python package is built. In our case, it contains configuration for Black (clang-format for Python) and pylint (clang-tidy) - with this file you can just run `black --check <filename>` instead of `black --check -l 99 -t py36 .... py311`.